File: app/src/main/java/dk/itu/moapd/viewpage/MainFragment.java
package dk.itu.moapd.viewpage;

import android.graphics.Color;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ScrollView;

import java.util.Random;

public class MainFragment extends Fragment {

    private int mColor;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_main, container, false);

        if (savedInstanceState == null) {
            Random random = new Random();
            mColor = Color.argb(255, random.nextInt(256),
                    random.nextInt(256),
                    random.nextInt(256));
        } else
            mColor = savedInstanceState.getInt("COLOR");

        ScrollView scrollView = view.findViewById(R.id.scroll_view);
        scrollView.setBackgroundColor(mColor);

        return view;
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt("COLOR", mColor);
    }

}

File: app/src/main/java/dk/itu/moapd/viewpage/MainFragment.kt
package dk.itu.moapd.viewpage

import android.graphics.Color
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_main.*
import java.util.*

class MainFragment : Fragment() {

    private var mColor: Int? = 0

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_main, container, false)

        mColor = if (savedInstanceState == null) {
            val random = Random()
            Color.argb(255, random.nextInt(256),
                random.nextInt(256),
                random.nextInt(256))
        } else
            savedInstanceState.getInt("COLOR")

        return view
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        scroll_view.setBackgroundColor(mColor!!)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putInt("COLOR", mColor!!)
    }

}
